.circleci/config.yml

```diff
@@ -164,6 +164,53 @@ jobs:
       - run: make -C build coverage
       - run: /bin/bash <(curl -s https://codecov.io/bash) || echo "Codecov did not collect coverage reports"
 
+  clang-build-debug:
```

Comment:
Made a related comment below: since we run clang tools as part of the `build-lint-debug` job, I would be in favor of turning that task into one using clang 10 on Linux and not adding two additional jobs. The current compiler used there is gcc 5.4.0, which is from ~5 years ago.

What do you think @SvetlanaBayarovich ?

Reply:
I had troubles with CI. My initial idea was that it was happening due to the lack of support on the side of `gcc`, that's why I was trying `clang`. ([details](https://github.com/valhalla/valhalla/pull/2487#discussion_r460789889))

Reply:
> I would be in favor of turning that task into one using clang 10 on Linux and not adding two additional jobs. The current compiler used there is gcc 5.4.0, which is from ~5 years ago.

We still want to build on both gcc and clang as I understand it. Why not enable more checks and sanitizers on the Osx build where we use clang already?

As for the old gcc, I think the better solution would be to migrate away from Ubuntu 16 as the base image.

Reply:
Well, I suggested using that particular task (`build-lint-debug`) since it was already a central point of using Clang tools for linting/debugging/etc. The other tasks are still using gcc. I'm also fine with enabling some of these checks in OSX as well. Upgrading Ubuntu is fine, but we should use a fixed, explicit, recent compiler version.

Regarding sanitizer support, we should prioritize getting sanitizer support functional on LLVM (vs their ports to GCC).

Reply:
It turned out that the issues I was having were related only to ASan+python/node bindings interaction (order of library loading). The current solution I suggest is to disable the bindings in the lint job and add sanitizers to it.

The bindings are still build e.g. in `build-release` job. This job has `-DENABLE_PYTHON_BINDINGS=On` and node bindings are `On` by default according to our CMakeLists. I can also add `-DENABLE_NODE_BINDINGS=On` to make it more explicit. WDYT?

Reply:
UPD ^^ as long as node bindings are removed the point relates only to python bindings